package/MDAnalysis/core/selection.py

```diff
@@ -537,6 +537,101 @@ def _apply(self, group):
         return group[np.asarray(indices, dtype=np.int64)]
 
 
+class BoxSelection(Selection):
+    token = "box"
+    precedence = 1
+    index_map = {"x": 0, "y": 1, "z": 2}
+    combination = [
+        "x",
+        "y",
+        "z",
+        "xy",
+        "xz",
+        "yz",
+        "yx",
+        "zx",
+        "zy",
+        "xyz",
+        "xzy",
+        "yxz",
+        "yzx",
+        "zxy",
+        "zyx",
+    ]
+    axis_map = ["x", "y", "z"]
+
+    def __init__(self, parser, tokens):
+        super().__init__(parser, tokens)
+        self.periodic = parser.periodic
+        self.direction = tokens.popleft()
+        if self.direction not in self.combination:
```

Comment:
can we set `xmin` `xmax` etc here to `None` then optionally set them to something else in the `else:` branch below?  I think this would simplify the `__getattribute__` usage below